Setup GitHub action for running tests

stephenmathieson opened the pull request:
I want to add the feature discussed in #2, but want to make sure I don't break anything by doing so. This patch adds a GitHub action for running the tests and updates the `docker-compose` configuration (which enables the tests to run/pass).

stephenmathieson committed:
Setup GitHub action for running tests

This patch adds a GitHub action for running the tests and updates the `docker-compose` configuration (which enables the tests to run/pass).

stephenmathieson commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,15 @@
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12'
```
Tests fail on Node 14. Haven't investigated why 🤷

stephenmathieson commented:
Looks like the DB isn't ready for connections yet. I'll take a look at it tomorrow morning.

nettofarah commented:
Anything I can do to help here, @stephenmathieson?
I think you can use something like [wait-port](https://www.npmjs.com/package/wait-port) to wait for the db connection before running the tests.

stephenmathieson commented:
Thanks for the invite (and `wait-port` tip)!

I ended up spending all day Friday fixing a pretty scary security issue for work and didn't find any time to finish this. I'll get to it in the next few days tho!

stephenmathieson committed:
Merge branch 'master' into setup-ci

stephenmathieson committed:
Refactor test script, add wait-port

stephenmathieson committed:
Disconnect from DB

To avoid seeing tons of "_Creating a duplicate database object for the same connection_" warnings, we're now disconnecting from the database after generating code.

stephenmathieson committed:
Revert "Disconnect from DB"

This reverts commit 4832356c70a5e15027591914bfa0591aa3ae782a. While the change prevented the warnings locally, it had no effect in CI :shrug:

stephenmathieson committed:
Refactor query method

Wait until we've connected before running a query

stephenmathieson committed:
Revert "Refactor query method"

This reverts commit ed25e3ffc4992a67a7662ff39f8f7fc168cf835f. I have no idea what's going on

stephenmathieson commented:
Messed around and tried a couple things, but had no luck. Not sure what's going on in CI. I'll take another stab at this in the future.
